blueocean-core-js/src/js/components/LoginButton.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component, PropTypes } from 'react';
+import {
+    AppConfig, Security, UrlConfig, Utils,
+} from '../index';
+
+/**
+ * LoginButton to login/logout.
+ */
+export class LoginButton extends Component {
+
+    render() {
+        if (!Security.isSecurityEnabled()) {
+            return (null);
+        }
+
+        let action;
+        let url;
+        if (Security.isAnonymousUser()) {
+            action = 'login';
+            url = `${UrlConfig.getJenkinsRootURL()}/${AppConfig.getLoginUrl()}?from=${encodeURIComponent(Utils.windowOrGlobal().location.pathname)}`;
+        } else {
+            action = 'logout';
+            url = `${UrlConfig.getJenkinsRootURL()}/logout`;
+        }
+        return (
+            <div className={ this.props.className }>
+                <a href={url} className="btn-link">
+                    { this.props.translate(action, { defaultValue: action }) }
+                </a>
+            </div>
+        );
+    }
+}
+
+LoginButton.propTypes = {
+    className: PropTypes.string,
+    translate: PropTypes.func,
```

Comment:
Should I use `t` as propery name for the translator? It seemed cryptically short, since a translator for a button isn't the most obvious property. But then again most (all?) other translator properties in the source code just use `t`.